src/authorization/authorization.ts

```diff
@@ -425,15 +429,58 @@ export function initialize(
     }
   };
 
+  const getMergeDefaultValue = (merge?: boolean) => {
+    const doesAnonUserExist = getAnonUserId() === null;
+    return merge === undefined
+      ? (authIdentifier === null && typeOfAuth === null && doesAnonUserExist) || // Criteria is not yet met (default merge is true)
+        (authIdentifier !== null && typeOfAuth !== null && !doesAnonUserExist) // Criteria is met (Iterable profile created with an autogenerated identity)(default merge is true)
+        ? true
+        : false // Current logged in user is identified (default merge is false)
+      : merge;
+  };
+
+  const handleMerge = async (
+    emailOrUserId: string,
```

Comment:
can we make this just take the same params as `tryMergeUser`? Then you won't need to do the conversions on line 447-452,

Reply:
sorry meant `mergeUser`, not `tryMergeUser`

Reply:
Actually we need to handle the case in tryMergeUser where we won't call  anonymousUserMerge.mergeUser at all and return promise with true, hence we need this additional layer.